oak_functions/loader/src/server.rs

```diff
@@ -384,14 +356,38 @@ impl WasmState {
                 OakStatus::ErrInvalidArgs
             });
 
+        // TODO(#2707): Refactor to return OakStatus::ErrInvalidHandle.
+        let extension = self
+            .extensions_indices
+            .iter_mut()
+            .find_map(|(_, extension)| {
+                if extension.get_handle() == handle {
+                    Some(extension)
+                } else {
+                    None
+                }
+            })
+            .expect("Fail to find extension with given handle.");
```

Comment:
I think this is covered by the TODO above, but I also agree this is super important, otherwise a Wasm module can invoke a non-existing extension in order to cause a panic in the entire Wasm execution, which may be externally observable.

Reply:
Good point, both! I did fix it immediately, but will iterate on the TODO.